agent/util-scripts/pbench-register-tool

```diff
@@ -175,6 +176,24 @@ while true; do
 		shift
 		_do_test_labels=1
 		;;
+	--persistent)
+		shift
+		display_name=${name}
+		name=$(${pbench_bin}/util-scripts/get-internal-tool ${pbench_bin}/tool-scripts ${name} persistent)
+		if [[ -z "${name}" ]]; then
+			name="internal-tool-not-found"
+		fi
+		type="persistent"
+		;;
+	--transient)
+		shift
+		display_name=${name}
+		name=$(${pbench_bin}/util-scripts/get-internal-tool ${pbench_bin}/tool-scripts ${name} transient)
+		if [[ ${name} == "" ]]; then
+			name="internal-tool-not-found"
+		fi
```

Comment:
Well, OK; unwieldy, and I always generally dislike repurposing a variable for error handling; but, fine.

Reply:
Oh wait a second, that should've also been replaced with a `-z "${name}"` in the condition...